libs/providers/flagsmith/src/lib/flagsmith-provider.ts

```diff
@@ -59,33 +108,42 @@ export default class FlagsmithOpenFeatureProvider implements Provider {
     try {
       const flags = await this.getFlags(evaluationContext);
       flag = flags.getFlag(flagKey);
-    } catch (ex) {
-      throw new FlagsmithProviderError(
-        'An error occurred retrieving flags from Flagsmith client.',
-        ErrorCode.GENERAL,
-      );
+    } catch (error) {
+      throw new FlagsmithProviderError('An error occurred retrieving flags from Flagsmith client.', ErrorCode.GENERAL);
     }
 
-    if (flag.isDefault && !this.useFlagsmithDefaults) {
+    // Do we want to raise an error if the flag is not found?
+    if (!this.useFlagsmithDefaults && (!flag || flag?.isDefault)) {
       throw new FlagNotFoundError(`Flag '${flagKey}' was not found.`);
     }
 
-    if (flagType === 'boolean' && !this.useBooleanConfigValue) {
-      return { value: flag.enabled };
+    if (!(this.returnValueForDisabledFlags || flag.enabled)) {
+      throw new FlagsmithProviderError(`Flag '${flagKey}' is not enabled.`, ErrorCode.GENERAL);
     }
 
-    if (!(this.returnValueForDisabledFlags || flag.enabled)) {
-      throw new FlagsmithProviderError(
-        `Flag '${flagKey}' is not enabled.`,
-        ErrorCode.GENERAL,
-      );
+    if (!this.useBooleanConfigValue && flagType === 'boolean') {
```

Comment:
I hesitated to turn those conditions into helpers, e.g `shouldUseBooleanConfig`. I don't know if it would really increase readability.
Keeping the configuration plain straight here helps understand how the config works.